app/src/main/java/com/epfl/drawyourpath/userProfile/UserModel.kt

```diff
@@ -0,0 +1,354 @@
+        this.database = database
+
+        //check the userId
+        if(userId.isEmpty()){
+            throw java.lang.Error("The userId can't be empty !")
+        }
+        if(!checkUserId(userId,database)){
+            throw java.lang.Error("The userId must be present on the database !")
+        }
+        this.userId=userId
+
+        //check the username
+        if(username.isEmpty()){
+            throw java.lang.Error("The username can't be empty !")
+        }
+        if(!checkUsername(userId,username, database)){
+            throw java.lang.Error("The username not correspond to the given userId !")
+        }
+        this.username=username
+
+        //check the format of the mail
+        if(!checkEMail(emailAddress)){
+            throw java.lang.Error("The mail address is not in the correct format !")
+        }else{
+            this.emailAddress = emailAddress
+        }
+
+        //check the format of the firstname
+        if(!checkName(firstname)){
+            throw java.lang.Error("Incorrect firstname")
+        }else{
+            this.firstname = firstname
+        }
+
+        //check the format of the surname
+        if(!checkName(surname)){
+            throw java.lang.Error("Incorrect surname")
+        }else {
+            this.surname = surname
+        }
+
+        //check that the birth date respect the age condition of the app(10<=age<=100)
+        if(!checkDateOfBirth(dateOfBirth)){
+            throw java.lang.Error("Incorrect date of birth !")
+        }else{
+            this.dateOfBirth=dateOfBirth
+        }
+
+        //test the goals, the goals can't be equal to 0
+        if(distanceGoal==0.0){
+            throw java.lang.Error("The distance goal can't be equal to 0.")
+        }
+        this.distanceGoal=distanceGoal
+        if(activityTimeGoal==0.0){
+            throw java.lang.Error("The activity time goal can't be equal to 0.")
+        }
+        this.activityTimeGoal=activityTimeGoal
+        if(nbOfPathsGoal==0){
+            throw java.lang.Error("The number of paths goal can't be equal to 0.")
+        }
+        this.nbOfPathsGoal=nbOfPathsGoal
+
+        this.friendsList = HashMap()
+        //TODO: in the next task, the firstname, surname, dateOfBirth and the goals will be set to the database here
+    }
+
+    /**
+     * Get the userId of the user
+     * @return userId of the user
+     */
+    fun getUserId(): String{
+        return userId
+    }
+
+    /**
+     * Get the username of the user
+     * @return username of the user
+     */
+    fun getUsername(): String{
+        return username
+    }
+
+    /**
+     * Use this function to modify the username(the username will be modify only if it is available on the database)
+     * @param username that we want to set
+     */
+    fun setUsername(username: String){
```

Comment:
We might want to know when the operation completes to show a loading screen to the user, so we should probably return a `Future<Boolean>`. This is the same for all setters of this class.